cmd/nitro/nitro.go

```diff
@@ -241,6 +241,10 @@ func mainImpl() int {
 	if nodeConfig.Execution.Sequencer.Enable && !nodeConfig.Execution.Sequencer.Dangerous.Timeboost.Enable && nodeConfig.Node.TransactionStreamer.TrackBlockMetadataFrom != 0 {
 		log.Warn("Sequencer node's track-block-metadata-from is set but timeboost is not enabled")
 	}
+	if nodeConfig.Node.EspressoCaffNode.Enable && (nodeConfig.Execution.Sequencer.Enable || nodeConfig.Node.Sequencer || nodeConfig.Node.SeqCoordinator.Enable || nodeConfig.Node.DelayedSequencer.Enable) {
```

Comment:
why are we adding conditions in two places?

Reply:
To have 2 layers of validation. This section checks in nitro.go which the docker container would be using as an entry point. It's also not very testable code. The config validation is actually testable and should give us an additional layer of defense against something incorrect happening in the event logic gets changed around in nitro.go

Reply:
It's also only possible for us to check for timeboost and the forwarding target here being turned on here (timeboost being enabled is also gated behind the sequencer being enabled I believe because it lives in the sequencer config)